pkg/core/pipeline/resource/main.go

```diff
@@ -0,0 +1,78 @@
+package resource
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/updatecli/updatecli/pkg/core/pipeline/scm"
+	"github.com/updatecli/updatecli/pkg/core/transformer"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/awsami"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/dockerdigest"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/dockerfile"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/dockerimage"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/file"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/githubrelease"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/gittag"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/helm"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/jenkins"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/maven"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/shell"
+	"github.com/updatecli/updatecli/pkg/plugins/resources/yaml"
+)
+
+type ResourceConfig struct {
+	DependsOn []string `yaml:"depends_on"`
+	Name      string
+	Kind      string
+	// Deprecated in favor of Transformers on 2021/01/3
+	Prefix string
+	// Deprecated in favor of Transformers on 2021/01/3
+	Postfix      string
+	Transformers transformer.Transformers
+	Spec         interface{}
+	// Deprecated field on version [1.17.0]
+	Scm   map[string]interface{}
+	SCMID string `yaml:"scmID"` // SCMID references a uniq scm configuration
+}
+
+// Unmarshal decode a source spec and returned its typed content
```

Comment:
```suggestion
// Unmarshal decode a source spec and return its typed content
```

Reply:
```suggestion
// Unmarshal decodes a source spec and returns its typed content
```

better with 3rd person